Source/ProgressionSystemRuntime/Private/Widgets/PSEndGameWidget.cpp

```diff
@@ -160,3 +165,28 @@ void UPSEndGameWidget::UpdateStarProgressBarValue(const FPoolObjectData& Created
 	UPSStarWidget& SpawnedWidget = CreatedData.GetChecked<UPSStarWidget>();
 	SpawnedWidget.UpdateProgressionBarPercentage(NewProgressBarValue);
 }
+
+// Updates the progression menu widget when player changed
+void UPSEndGameWidget::UpdateProgressionWidgetForPlayer()
+{
+	UPSSaveGameData* SaveGameData = UPSWorldSubsystem::Get().GetCurrentSaveGameData();
+	if (!SaveGameData)
+	{
+		return;
+	}
+
+	const FPSSaveToDiskData& CurrenSaveToDiskDataRow = UPSWorldSubsystem::Get().GetCurrentSaveToDiskRowByName();
+	const FPSRowData& CurrenProgressionSettingsRow = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName();
```

Comment:
Please populate `CurrenSaveToDiskDataRow` and `CurrentProgressionSettingsRow` as function arguments of the delegate, instead of obtaining them directly from Getter, as their usage looks useful I suspect those arguments will be needed for any callback that is bound to this new `OnCurrentScoreChanged` delegate.

It shouldn't be problem for delegate to provide such arguments, since such data should be guaranteed to be available when scores are changed. If save is not created, then `OnScoreChanged` delegate shouldn't be called, instead system will use `OnInitialize` delegate

Reply:
done: https://github.com/h4rdmol/ProgressionSystem/pull/45/commits/2066d6cd859082853d388c78033f11867de0a025